### PR TITLE
Keep password order

### PR DESCRIPTION
**Pull request by puzzledsab**
Issue #1898

### CODE REVIEW

File: sabnzbd/misc.py [hunk 1]
```diff
@@ -779,7 +779,12 @@ def get_all_passwords(nzo):
         # If we're not sure about encryption, start with empty password
         # and make sure we have at least the empty password
         passwords.insert(0, "")
-    return set(passwords)
+
+    unique_passwords = []
+    for password in passwords:
+        if password not in unique_passwords:
+            unique_passwords.append(password)
+    return unique_passwords
```
Comment (Safihre) [inline sabnzbd/misc.py:787]: Wouldn't a nice 1 liner list comprehension be better here?

Comment (puzzledsab) [reply sabnzbd/misc.py:787]: Is that possible when it's comparing to the list it is generating? I don't know how.

Comment (JackDandy) [reply sabnzbd/misc.py:787]: perhaps...
```
>>> passwords = ['b', 'c', 'a', 'a', 'b']
>>> passwords
['b', 'c', 'a', 'a', 'b']
>>> unique_passwords = list(dict((pw, None) for pw in passwords))
>>> unique_passwords
['b', 'c', 'a']
```
py3 dict type is ordered so using the keys of a dict is an ordered set, and fast. ;)

Comment (Safihre) [reply sabnzbd/misc.py:787]: Oh yes sorry, can't tdo this really in a oneliner. Fine this way.

**Pull request merged**